dfastmi/gui.py

```diff
@@ -315,10 +316,10 @@ def updated_branch(ibranch: int) -> None:
         Newly selected branch number.
     """
     reach = dialog["reach"]
-    reach.clear()
-    reach.addItems(rivers.allreaches[ibranch])
+    reach.clear()    
+    reach.addItems([reach.name for reach in rivers.branches[ibranch].reaches])
```

Comment:
seems like you have become a python expert, since you have started using the list comprehension :man_student:

Reply:
I agree, but gui.py is not in scope of this issue. I only refactor code to get it working again. Will pickup the good comments later in new issue

Reply:
so you agree that you are now a python expert :camera_flash:

Reply:
hahahaha, maybe. Maybe not a python expert, but I do know how to program

Reply:
Discussed, we will leave these comments for a later refactoring.
When a new issue is created for this, the number will be disclosed here.